### ISSUE TITLE
GeolocateControl options hard to follow

### ISSUE THREAD

**Issue by stevage** [mapbox/mapbox-gl-js-docs#277]
While answering [this question](https://stackoverflow.com/questions/61194996/prevent-geolocatecontrol-zoom-out/61220862#61220862) I realised just how hard it is to follow the documentation for the various options on GeolocateControl.

There's a lot going on:

- options applied directly to the constructor (`showAccuracyCircle` for instance)
- a `positionOptions` object, whose options are determined by an external API
- a `fitBoundsOptions` object, whose options are defined by the `fitBounds` object, or by `AnimationOptions` or `CameraOptions`. (And it's quite confusing that "fitBounds" is mentioned here, when there's not really any obvious connection with fitting any bounds...)

It's really quite messy and confusing, both to remember what level of nesting/sub-object is required, and to understand that options from three different places can be combined in the `fitBoundsOptions` part.

I'm not sure I have any great suggestions, but I can say that sometimes documentation in the form of pseudo-Javascript can be much more readable. In this case it would look like:

```js
new GeolocateControl({
   positionOptions: {  // OPTIONAL
     enableHighAccuracy: true, // etc, see https://developer.mozilla.org/en-US/docs/Web/API/PositionOptions
   },
   fitBoundsOptions: { // OPTIONAL
     padding: 30, // etc, see https://docs.mapbox.com/mapbox-gl-js/api/#map#fitbounds
     animate: false, // all AnimateOptions are supported
     zoom: 15, // all CameraOptions are supported
   },
   trackUserLocation // description...
   ...
});
```  

Using the word "Object" is much more abstract than simply showing the `{ }` notation, and is harder on users who don't necessarily know the official JavaScript terminology. (I regularly encounter people who call them hashes, for instance.)

**Comment by katydecorah** [comment]
@stevage we're working on updating documentation for GeolocateControl right now in https://github.com/mapbox/mapbox-gl-js/pull/9607/files#diff-8c498cf163efa60ad84199c531ab7c5a want to take a look?

**Comment by stevage** [comment]
It's good that you're including example code (although, still pre ES2015, sigh...), but IMHO you can do better than:

```
geolocate.on('trackuserlocationend', function() {
   console.log('A trackuserlocationend event has occurred.')
 });
```

Including a bunch of examples that are the same code with two lines changed in the same way looks like it might just become annoying noise. If there aren't resources to develop meaningful examples for each event (elucidating *why* a user might want to track this particular event), then maybe just a single example for event handling is enough. Or perhaps I'm misunderstanding what this will look like in the final product.

**Comment by andrewharvey** [comment]
> a fitBoundsOptions object, whose options are defined by the fitBounds object, or by AnimationOptions or CameraOptions. (And it's quite confusing that "fitBounds" is mentioned here, when there's not really any obvious connection with fitting any bounds...)

I agree, it's referring to an implementation detail internal to the control (`fitBounds`), a term which is some combination of Camera and Animation options is probably better here. It's possible to change this without causing a breaking change by still supporting `fitBoundsOptions`

> It's really quite messy and confusing, both to remember what level of nesting/sub-object is required, and to understand that options from three different places can be combined in the fitBoundsOptions part.

I don't see any better alternatives, each of these options controls different things and the values permitted are set as you said externally.

> Using the word "Object" is much more abstract than simply showing the { } notation, and is harder on users who don't necessarily know the official JavaScript terminology. (I regularly encounter people who call them hashes, for instance.)

That's a wider issue than just with the GeolocateControl, it's across the board in the GL JS docs, better to cut that out into a standalone ticket.

> I'm not sure I have any great suggestions, but I can say that sometimes documentation in the form of pseudo-Javascript can be much more readable. In this case it would look like:

We do already have one example here, generally though most people will just copy paste that example, so I think it should have reasonable defaults at least for the first example, which means we probably shouldn't include fitBoundsOptions or positionOptions as they already have good defaults.
